### PR TITLE
Make game_handler.py work with `zulip-terminal`

### LINKED ISSUE #315: Make game_handler work using zulip_terminal
Currently **tictactoe** and **connect_fout** bots are not working using the `zulip_terminal <bot-name>` method.
It raises following error for `$ zulip_terminal tictactoe`:
```
Traceback (most recent call last):
  File "/home/a/python-zulip-api/zulip-api-py3-venv/bin/zulip-terminal", line 11, in <module>
    load_entry_point('zulip-bots', 'console_scripts', 'zulip-terminal')()
  File "/home/a/python-zulip-api/zulip_bots/zulip_bots/terminal.py", line 57, in main
    message_handler.initialize(bot_handler)
  File "/home/a/python-zulip-api/zulip_bots/zulip_bots/game_handler.py", line 137, in initialize
    self.email = self.bot_handler.email
AttributeError: 'TerminalBotHandler' object has no attribute 'email'
```

### PR DESCRIPTION
**Pull request by amanagr**
Make zulip_bots.zulip_bots.terminal.py work with game_handler.py
Add conf info for tictactoe and connect_four bots
Fixes #315

### CODE REVIEW

File: zulip_bots/zulip_bots/terminal.py [hunk 1]
```diff
@@ -57,15 +57,20 @@ def main():
         message_handler.initialize(bot_handler)
 
     sender_email = 'foo_sender@zulip.com'
+    sender_full_name = 'foo'
+    message_type = 'stream'
+    subject = 'testplay'
```
Comment (roberthoenig) [inline zulip_bots/zulip_bots/terminal.py]: How about just "test" for the subject?

File: zulip_bots/zulip_bots/game_handler.py [hunk 2]
```diff
@@ -134,8 +134,9 @@ def usage(self) -> str:
     def initialize(self, bot_handler: Any) -> None:
         self.bot_handler = bot_handler
         self.get_user_cache()
-        self.email = self.bot_handler.email
-        self.full_name = self.bot_handler.full_name
+        self.config_info = bot_handler.get_config_info('tictactoe')
+        self.email = self.config_info.get("email", False)
+        self.full_name = self.config_info.get("full_name", False)
```
Comment (roberthoenig) [inline zulip_bots/zulip_bots/game_handler.py]: The original code looks correct here and should remain unchanged. Rather, we probably need to update the TerminalBotHandler class in simple_lib.py (in specific, the __init__ method).

File: zulip_bots/zulip_bots/game_handler.py [hunk 3]
```diff
@@ -646,7 +647,7 @@ def get_game_id_by_email(self, user_email: str) -> str:
         return ''
 
     def get_bot_username(self) -> str:
-        return self.bot_handler.full_name
+        return self.full_name
```
Comment (roberthoenig) [inline zulip_bots/zulip_bots/game_handler.py]: This change looks correct, but doesn't seem to be needed for this commit. If you want, you can make a new commit with it.

File: zulip_bots/zulip_bots/bots/tictactoe/tictactoe.conf [hunk 4]
```diff
@@ -0,0 +1,3 @@
+[tictactoe]
+email=boo_sender@zulip.com
+full_name=foo
```
Comment (roberthoenig) [inline zulip_bots/zulip_bots/bots/tictactoe/tictactoe.conf]: The tictactoe bot doesn't require a conf file. Rather, `email` and `full_name` are normally read from a bot's zuliprc file. See my next comment on how  to make this work with the terminal script.

File: zulip_bots/zulip_bots/bots/tictactoe/test_tictactoe.py [hunk 5]
```diff
@@ -65,5 +71,6 @@ def setup_game(self) -> None:
         return bot
 
     def _get_game_handlers(self) -> Tuple[Any, Any]:
-        bot, bot_handler = self._get_handlers()
+        with self.mock_config_info(self.mock_config):
+            bot, bot_handler = self._get_handlers()
```
Comment (roberthoenig) [inline zulip_bots/zulip_bots/bots/tictactoe/test_tictactoe.py]: There shouldn't be any changes to the Tictactoe bot - see comments below.

**Comment by amanagr** [comment]
Test for bots.baremetrics.test_baremetrics is failing on travic-ci but passes on my local computer.
I check tests via `./tools/test-bots`.

File: zulip_bots/zulip_bots/simple_lib.py [hunk 6]
```diff
@@ -35,6 +35,9 @@ def __init__(self, bot_config_file):
         self.bot_config_file = bot_config_file
         self.storage = SimpleStorage()
         self.message_server = SimpleMessageServer()
+        if self.bot_config_file is None:
```
[the remark this answers is not in the document]
Comment (showell) [reply zulip_bots/zulip_bots/simple_lib.py]: yep, I don't think we need to check for bot_config_file here

**Comment by showell** [comment]
@amanagr We can probably make this try to support multiple emails somehow, perhaps by having a pseudocommand like "/menu" that presents more options.

**Comment by amanagr** [comment]
I have found an easy fix for 2-player games, since we don't have any games which support more than two players at one, this fix shouldn't be a problem( atleast for testing purposes ).
Here is a demo:
Enter your message:
```
start game
join -cp
move 1
move 2 -cp
...
```
I will update the documentation if you approve of this fix.
@showell

**Comment by amanagr** [comment]
I don't really understand what you mean by `adding a pseudo command like "/menu"`, if can explain it, I can work on implementing it too.

**Comment by showell** [comment]
Don't worry about "/menu" yet, because I like the direction of your workaround.  I'm not sure exactly what "-cp" stands for, but I get the gist of what you're doing.  Essentially, we want some way for the sender to take on the role of the second player.  One thing that happens in the email world is that most email systems let you put something "+test" after your name in the email, so that people can associate multiple email addresses (from the "outside" perpsective) with a single actual email account.

I'm wondering if there's a more elegant/obvious way to express this for the game player.

**Comment by amanagr** [comment]
I think this fix will easily allow multiplayer support in zulip-terminal.
Demo:
```
start game
join -p a
join -p b
move 1
move 2 -p a
move 3 -p b
...
```
I think this is what you were trying to say.

**Comment by amanagr** [comment]
Closing since they changes here are easy to write or get back to when we want to do this.

**Pull request closed without merging**